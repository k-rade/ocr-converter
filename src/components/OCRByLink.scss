.ocr-by-link {
  width: 100%;
  display: flex;
  flex-direction: column;
  margin: 1rem 0 0.5rem;

  &-input {
    font-family: var(--font);
    color: var(--color-white);
    background-color: var(--color-gray-800);
    border: none;
    border-radius: 2px;
    padding: 0.625rem 1rem;
    outline: none;
    font-size: 0.875rem;
    caret-color: var(--color-primary-600);
    margin: 0 0 1rem;

    &::placeholder {
      color: var(--color-gray-200);
    }

    &:focus {
      box-shadow: 0 0 2px var(--color-gray-200) inset;
    }
  }
}