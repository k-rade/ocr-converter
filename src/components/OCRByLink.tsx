import ImageWrapper from "./ImageWrapper";

import "./OCRByLink.scss";

interface Props {
  setImage: (value: string) => void;
  image: string;
}

const OCRByLink = ({ setImage, image }: Props): JSX.Element => {
  return (
    <div className="ocr-by-link">
      <input
        type="text"
        placeholder="Enter image URL ..."
        className="ocr-by-link-input"
        value={image}
        onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
          setImage(e.target.value)
        }
      />
      <ImageWrapper imageUrl={image} />
    </div>
  );
};

export default OCRByLink;
