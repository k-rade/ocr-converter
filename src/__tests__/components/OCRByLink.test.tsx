import { fireEvent, render } from "@testing-library/react";
import OCRByLink from "../../components/OCRByLink";

describe("OCRByLink", (): void => {
  it("should call set image", (): void => {
    const mockSetImage = jest.fn();
    const { getByTestId } = render(
      <OCRByLink image="image url" setImage={mockSetImage} />
    );
    fireEvent.change(getByTestId("ocr-image-link"), {
      target: { value: "new image url" },
    });

    expect(mockSetImage).toHaveBeenCalledWith("new image url");
  });

  it("should render", (): void => {
    const { baseElement } = render(
      <OCRByLink image="image url" setImage={jest.fn} />
    );

    expect(baseElement).toMatchSnapshot();
  });
});
