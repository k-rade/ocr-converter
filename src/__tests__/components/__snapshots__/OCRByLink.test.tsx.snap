// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OCRByLink should render 1`] = `
<body>
  <div>
    <div
      class="ocr-by-link"
    >
      <input
        class="ocr-by-link-input"
        data-testid="ocr-image-link"
        placeholder="Enter image URL ..."
        type="text"
        value="image url"
      />
      <div
        class="image-wrapper"
      >
        <img
          alt="can not load image"
          src="image url"
        />
      </div>
    </div>
  </div>
</body>
`;
